src/client/client/game.js

```diff
@@ -139,6 +150,121 @@ let game = (function() {
+				if (!isInitialState) {
+					_audio.end.play();
+
+					_DOMElements.raceLeaderboard.className = "visible";
+
+					_enteredMarbleList.sort((a, b) => {
+						if (a.finished && b.finished) {
+							return b.time - a.time;
+						} else if (a.finished) {
+							return 1;
+						} else if (b.finished) {
+							return -1;
+						} else {
+							return 0;
+						}
+					});
+
+					// Reverse it cause im stupid
+					_enteredMarbleList.reverse();
+
+					_DOMElements.resultsList.innerHTML = "";
+
+					// Get current user_data to see if any row needs to be highlighted
+					let user_data = Cookies.getJSON("user_data");
+
+					// Build leaderboard DOM
+					let resultsListFragment = new DocumentFragment();
+					for (let i = 0; i < _enteredMarbleList.length; i++) {
+						let marble = _enteredMarbleList[i];
+						let resultsEntry = _DOMElements.resultsListTemplate.cloneNode(true);
+
+						resultsEntry.removeAttribute("id");
+
+						// Highlight for current player
+						if (user_data && user_data.id === marble.userId) {
+							resultsEntry.className += " currentPlayer";
+						}
+
+						// Rank & name
+						resultsEntry.getElementsByClassName("rank")[0].innerText = (i + 1);
+						resultsEntry.getElementsByClassName("name")[0].innerText = marble.name;
+
+						// Marble finished state
+						if (marble.finished) {
+							resultsEntry.getElementsByClassName("time")[0].innerText = (marble.time * .001).toFixed(2);
+							resultsEntry.getElementsByClassName("timediff")[0].innerText = i !== 0
+								? `+${((marble.time - _enteredMarbleList[0].time) * .001).toFixed(2)}`
+								: "-";
+						} else {
+							resultsEntry.getElementsByClassName("time")[0].className += " dnf";
+						}
+
+						// In case our marble is human
+						if (marble.userId) {
+							// PBs
+							if (newStateData.data[marble.userId].record === "pb") {
+								resultsEntry.getElementsByClassName("record")[0].innerText = "PB";
+								resultsEntry.getElementsByClassName("record")[0].className += " pb";
+							}
+
+							// Points earned this round
+							resultsEntry.getElementsByClassName("points")[0].innerText = `+${newStateData.data[marble.userId].pointsEarned}`;
+
+							// Points earned over time
+							resultsEntry.getElementsByClassName("pointstotal")[0].innerText = newStateData.data[marble.userId].pointsTotal;
+						} else {
+							// Points earned this round
+							resultsEntry.getElementsByClassName("points")[0].className += " none";
+						}
+
+						resultsListFragment.appendChild(resultsEntry);
+					}
+					_DOMElements.resultsList.appendChild(resultsListFragment);
+
+					_DOMElements.raceLeaderboardMapName.innerText = newStateData.data.map.name;
+					_DOMElements.raceLeaderboardAuthorName.innerText = newStateData.data.map.author;
+
+					// Automatic scroll
+					if (_DOMElements.resultsList.scrollHeight > 0) {
+						_DOMElements.resultsList.scrollTop = _DOMElements.resultsList.scrollHeight;
+
+						// Start scrolling up
```

Comment:
It seems more sensible to scroll from top to bottom. Seeing top ranks first is more interesting than the late/DNF'd marbles. (Or maybe allow the user to scroll through?)

Reply:
Scrolling will be removed and only the top ranking entries will be shown by default. The user can scroll manually through the list as long as it is on screen. This is done as an incentive to play more and reach a higher rank, to claim a few seconds of fame.